### PR TITLE
Add method to fetch machine image information

### PR DESCRIPTION
**Pull request by cjh1**

### CODE REVIEW

**Comment by cjh1** [comment]
@kotfic PTAL

File: cumulus/ansible/tasks/providers/ec2.py [hunk 1]
```diff
@@ -187,6 +187,15 @@ def get_volume(self, girder_volume):
...
         return volume
 
+    def get_machine_images(self, name, owner):
+        filters = [{
+            'Name': 'name',
+            'Values': [name]
+        }]
+        r = self.client.describe_images(Owners=[owner], Filters=filters)
+
+        return [{'image_id': image['ImageId']} for image in r['Images']]
```
Comment (cjh1) [inline cumulus/ansible/tasks/providers/ec2.py]: @kotfic I used underscores here to be consistent with the rest of the class. However, this seems to do against the camel case convention?

Comment (kotfic) [inline cumulus/ansible/tasks/providers/ec2.py]: @cjh1 It looks like Owner can be added to the filters argument passed to to [describe_images](http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.Client.describe_images).  How do you feel about changing the signature of this function to  ```def get_machine_images(self, filters=None)``` 

That way other cloud providers can implement their own logic for filters and (in the future)  we can refactor this functionality into an internal function and build a provider specific mapping (e.g.  ```provider.get_image('ubuntu', '16.04')```)

**Comment by cjh1** [comment]
@kotfic PTAL

**Comment by kotfic** [comment]
@cjh1 LGTM!

**Pull request merged**